lib/_http_server.js

```diff
@@ -161,46 +161,52 @@ ServerResponse.prototype._implicitHeader = function _implicitHeader() {
 
 ServerResponse.prototype.writeHead = writeHead;
 function writeHead(statusCode, reason, obj) {
-  var headers;
+  statusCode |= 0;
+  if (statusCode < 100 || statusCode > 999)
+    throw new RangeError(`Invalid status code: ${statusCode}`);
 
   if (typeof reason === 'string') {
     // writeHead(statusCode, reasonPhrase[, headers])
     this.statusMessage = reason;
   } else {
     // writeHead(statusCode[, headers])
-    this.statusMessage =
-        this.statusMessage || STATUS_CODES[statusCode] || 'unknown';
+    if (!this.statusMessage)
+      this.statusMessage = STATUS_CODES[statusCode] || 'unknown';
     obj = reason;
   }
   this.statusCode = statusCode;
 
+  var headers;
   if (this._headers) {
     // Slow-case: when progressive API and header fields are passed.
+    var k;
     if (obj) {
       var keys = Object.keys(obj);
       for (var i = 0; i < keys.length; i++) {
-        var k = keys[i];
+        k = keys[i];
```

Comment:
It wasn't about performance but easier readability and because I check if the loop was entered below by checking `k`.